action/kafkaProduce.py

```diff
@@ -1,4 +1,5 @@
 import ssl
+import base64
 from kafka import KafkaProducer
 from kafka.errors import NoBrokersAvailable
 
```

Comment:
@dubeejw We just went through this Python 2/3 bytes/text mess in actionProxy.  I think we can make this code work for both versions of Python.

In Python 2, `bytes` is a subset of `str`, and when you decode a `bytes/str` object, you get a unicode string. 

```
Python 2.7.13
>>> import base64
>>> winter64 = '4p2EIOKYgyDinYQgYnJyCg=='
>>> winter = base64.b64decode(winter64)
>>> winter
'\xe2\x9d\x84 \xe2\x98\x83 \xe2\x9d\x84 brr\n'
>>> type(winter)
<type 'str'>
>>> isinstance(winter, bytes)
True
>>> isinstance(winter, str)
True
>>> x = winter.decode('utf-8')
>>> type(x)
<type 'unicode'>
>>> isinstance(x, bytes)
False
>>> x
u'\u2744 \u2603 \u2744 brr\n'
>>> print(x)
❄ ☃ ❄ brr
```

In Python 3, `bytes` is binary data, `str` is a unicode string, and there is no `unicode` type.   So the types are a bit different, but the same code will give you the same output.

```
Python 3.5.4
>>> import base64
>>> winter64 = '4p2EIOKYgyDinYQgYnJyCg=='
>>> winter = base64.b64decode(winter64)
>>> winter
b'\xe2\x9d\x84 \xe2\x98\x83 \xe2\x9d\x84 brr\n'
>>> type(winter)
<class 'bytes'>
>>> isinstance(winter, bytes)
True
>>> isinstance(winter, str)
False
>>> x = winter.decode('utf-8')
>>> type(x)
<class 'str'>
>>> isinstance(x, bytes)
False
>>> x
'❄ ☃ ❄ brr\n'
>>> print(x)
❄ ☃ ❄ brr
```

It looks like the program is receiving base64 encoded strings, decoding to bytes, decoding the bytes to unicode strings, and finally, encoding the strings back to bytes before sending across the wire.

If the bytes to text conversion is needed, this works the same in both versions of Python:

```
>>> import base64
>>> winter64 = '4p2EIOKYgyDinYQgYnJyCg=='
>>> winter = base64.b64decode(winter64)
>>> x = winter.decode('utf-8')
>>> bytes(x.encode('utf-8'))
b'\xe2\x9d\x84 \xe2\x98\x83 \xe2\x9d\x84 brr\n'
```

If decoding bytes to text and back to bytes is not needed, then this works in both versions of Python:

```
>>> import base64
>>> winter64 = '4p2EIOKYgyDinYQgYnJyCg=='
>>> winter = base64.b64decode(winter64)
>>> bytes(winter)
b'\xe2\x9d\x84 \xe2\x98\x83 \xe2\x9d\x84 brr\n'
```

I know, it's clear as mud.  Basically, do you need to use the `decode` after `b64decode`?  If so, just invoke `encode` on the string object before sending to `bytes()`.